### PR TITLE
Clang-format our Game/Input code

### PR DESCRIPTION
**Pull request by garbear**
## Description

This is a follow-up to https://github.com/xbmc/xbmc/pull/17922, five months later. Some comment rules in `.clang-format` have been updated, causing a lot of new changes.

Like https://github.com/xbmc/xbmc/pull/17922, this concerns four code areas:

* `xbmc/cores/RetroPlayer` (no changes were needed)
* `xbmc/games`
* `xbmc/input`
* `xbmc/peripherals`

This PR was far from automated. Most of the work was moving comments to precede the declarations.

### Example

Current:

```c++
  m_iExitCode = EXITCODE_QUIT;
  m_bIsMuted = false; //! @todo fetch the correct initial value when system audiostatus is
                      //! implemented in libCEC
  m_bGoingToStandby = false;
```

Applied by clang-format:

```c++
  m_iExitCode = EXITCODE_QUIT;
  m_bIsMuted = false; //! @todo fetch the correct initial value when system audiostatus is
      //! implemented in libCEC
  m_bGoingToStandby = false;
```

My preference:

```c++
  m_iExitCode = EXITCODE_QUIT;

  //! @todo fetch the correct initial value when system audiostatus is
  //! implemented in libCEC
  m_bIsMuted = false;

  m_bGoingToStandby = false;
```

This is an example where clang-format improves readability, but a more human touch can produce better results.

## Motivation and Context

Make it easier to rebase GSoC projects (https://github.com/garbear/xbmc/pull/86, https://github.com/garbear/xbmc/pull/114 and https://github.com/garbear/xbmc/pull/120) so they can be eventually upstreamed.

## How Has This Been Tested?

Build on X11 succeeds.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

**Comment by garbear** [comment]
Note to self: replace `//@{` with `///@{`

**Comment by garbear** [comment]
Updated with a spelling fix.

I observe an advantage of this PR: With the comments preceding the definition, we can make cosmetic changes in the comments without affecting git blame of the code lines.

**Comment by DaveTBlake** [comment]
Just eye-balling this looks fine to me - all layout changes - but be great if @Rechi or @AlwinEsch could confirm in time for imminent alpha3

File: xbmc/input/actions/ActionIDs.h [hunk 1]
```diff
@@ -32,132 +32,192 @@
...
+#define REMOTE_1 59 //!< For example in videoFullScreen.xml window id=2005 you can
+#define REMOTE_2 60 //!< Enter time (mmss) to jump to particular point in the movie
 #define REMOTE_3 61
-#define REMOTE_4 62 //!< with spincontrols you can enter 3digit number to quickly set
+#define REMOTE_4 62 //!< With spincontrols you can enter 3digit number to quickly set
 #define REMOTE_5 63 //!< spincontrol to desired value
 #define REMOTE_6 64
 #define REMOTE_7 65
 #define REMOTE_8 66
 #define REMOTE_9 67
```
Comment (AlwinEsch) [inline xbmc/input/actions/ActionIDs.h:182]: Think to use here as:
```cpp
//! Remote keys 0-9. are used by multiple windows.
//!
//! For example in videoFullScreen.xml window id=2005 you can
//! Enter time (mmss) to jump to particular point in the movie
//!
//! With spincontrols you can enter 3digit number to quickly set
//! spincontrol to desired value.
//!@{
#define REMOTE_0 58

//! @see REMOTE_0 about details.
#define REMOTE_1 59

//! @see REMOTE_0 about details.
#define REMOTE_2 60

//! @see REMOTE_0 about details.
#define REMOTE_3 61

//! @see REMOTE_0 about details.
#define REMOTE_4 62

//! @see REMOTE_0 about details.
#define REMOTE_5 63

//! @see REMOTE_0 about details.
#define REMOTE_6 64

//! @see REMOTE_0 about details.
#define REMOTE_7 65

//! @see REMOTE_0 about details.
#define REMOTE_8 66

//! @see REMOTE_0 about details.
#define REMOTE_9 67
//!@}
```

With a `//!<` becomes everything added independently to his define.

Here how looks:
Before:
Bildschirmfoto vom 2020-10-29 20-01-37

---------------------------------------------

After:
Bildschirmfoto vom 2020-10-29 20-13-24

**Review by AlwinEsch — COMMENTED** [review]
Looking good to me and makes it much cleaner as the line breaks with clang are much cleaner.

**Comment by AlwinEsch** [comment]
> Note to self: replace `//@{` with `///@{`

Yes as there it no more works since last doxygen versions,
Also can be `//!@{` or `/*!@{*/`

**Comment by garbear** [comment]
@AlwinEsch thanks for noticing the opportunity to improve REMOTE_X documentation. I applied the change you suggested.

Jenkins is a bit backed up, but I plan to merge tomorrow morning.

**Pull request merged**